tensorflow_probability/python/experimental/mcmc/streaming_reductions_test.py

```diff
@@ -122,6 +125,32 @@ def test_calibration_warning(self):
         any('supplied `TransitionKernel` is not calibrated.' in str(
             warning.message) for warning in triggered))
 
+  def test_seed_reproducibility(self):
+    first_fake_kernel = RandomTransitionKernel()
+    last_state = self.evaluate(step_kernel(
+        num_steps=1,
+        current_state=0,
+        kernel=first_fake_kernel,
+        seed=test_util.test_seed(),
```

Comment:
There should be a single `seed=test_util.test_seed()` at the top, and the test should reuse that seed in each of these three places.  The reason is that when testing robustness to seed changes with `--test_arg=--vary_seed`, each call to `test_util.test_seed()` will actually return a different seed.

Reply:
When fixing the `seed` structure, I actually noticed that running that procedure in eager mode doesn't lead to deterministic results. Even when seeded, `first_fake_kernel` and `second_fake_kernel` give different results after `num_steps` steps. After some digging, I discovered the cause was because the seed was changing after calls to `samplers.sanitize_seed`. After subsequently looking through the tests for `samplers`, I also realized this is expected behaviour for eager mode.

Though the fix is rather simple, those realizations led me to also add a seed reproducibility test to `sample_chain` (since none of a similar flavor have been written yet), which also failed at first. I added all the changes I made to this PR (for reference), though the `sample_chain` edits should maybe be moved to another PR if it's actually a bug.

I'm wondering if what I found is actually expected behaviour (i.e. the user has to somehow find a way to seed if they're running eager mode, and it's not the responsibility of the driver), or if it's actually an oversight.